### ISSUE TITLE
wishlist: completion, function argument hints

### ISSUE THREAD

**Issue by tpapp** [stan-dev/stan-mode#7]
It would be nice to have
1. completion for functions (mostly distributions),
2. some kind of hints that could either be shown with a command, or automatically, in the minibuffer or as a popup hint. For example, for `normal(...)` it would display something like `normal(mu, sigma)`. Since STAN sometimes uses different semantics for functions that are also present in R etc (eg in R the `normal` uses variance), this could help prevent some coding errors.

I don't know how to implement these features in Emacs, but once a framework is in place, I would be happy to fill it in the actual definitions.
